core/lib/server/render/pub/renderWithPubUtils.ts

```diff
@@ -61,19 +82,62 @@ const getPubValue = (context: RenderWithPubContext, fieldSlug: string, rel?: str
 	return expect(pubValue, `Expected pub to have value for field "${fieldSlug}"`);
 };
 
-export const renderFormInviteLink = async ({
-	formSlug,
-	userId,
-	communityId,
-	pubId,
-}: {
-	formSlug: string;
-	userId: UsersId;
-	communityId: CommunitiesId;
-	pubId: PubsId;
-}) => {
-	await addMemberToForm({ userId, communityId, pubId, slug: formSlug });
-	return createFormInviteLink({ userId, formSlug, communityId, pubId });
+export const renderFormInviteLink = async (
+	{
+		formSlug,
+		recipient,
+		communityId,
+		pubId,
+		inviter,
+	}: {
+		formSlug: string;
+		recipient: Recipient;
+		communityId: CommunitiesId;
+		pubId: PubsId;
+		inviter: XOR<{ userId: UsersId }, { actionRunId: ActionRunsId }>;
+	},
+	trx = db
+) => {
+	// this feels weird to do here
+	if (recipient.id) {
+		await addMemberToForm({ userId: recipient.user.id, communityId, pubId, slug: formSlug });
+		return createFormInviteLink(
+			{ userId: recipient.user.id, formSlug, communityId, pubId },
+			trx
+		);
+	}
```

Comment:
this could be replaced with invites too if invites can act as "pseudo-users" and be selected on the "MemberSelect" dropdown